api/src/main/java/org/infinispan/api/async/AsyncCaches.java

```diff
@@ -0,0 +1,27 @@
+package org.infinispan.api.async;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Flow;
+
+import org.infinispan.api.configuration.CacheConfiguration;
+
+/**
+ * @since 14.0
+ **/
+public interface AsyncCaches {
+   <K, V> CompletionStage<AsyncCache<K, V>> create(String name, CacheConfiguration cacheConfiguration);
```

Comment:
We should define what happens if the cache already has a config and what happens if they are not the same.

Reply:
And do these methods really need to return a `CompletionStage`? The  cache itself is already async so I would say we don't need it here.